vowpalwabbit/core/include/vw/core/example_predict.h

```diff
@@ -42,6 +42,8 @@ class example_predict
   example_predict(example_predict&& other) = default;
   example_predict& operator=(example_predict&& other) = default;
 
+  void calculate_feature_space_hash();
```

Comment:
Can you make `feature_space_hash` private and make this a lazily evaluated value?